### ISSUE TITLE
Comma spacing issue with JSX

### ISSUE THREAD

**Issue by ustun** [babel/babel-eslint#18]
I get a warning with commas in JSX:
```
      413:72  error  A space is required after ','            comma-spacing
```
An example:
```
var X = React.createClass({

    render: function () {
        var hello = "Hello";
        var world = "World";
        return <span>{hello}, {world}</span>;
    }

});

```

**Comment by ustun** [comment]
A workaround for now is to use {','} instead.

**Comment by nzakas** [comment]
Can you see of you get the same error using the default ESLint parser? If so, please file a bug on ESLint.

**Comment by sebmck** [comment]
I'm seeing the same error when using the default ESLint parser, please report this upstream.

**.eslintrc**

```json
{
  "ecmaFeatures": {
    "jsx": true
  }
}
```

**index.js**

```javascript
var X = React.createClass({

    render: function () {
        var hello = "Hello";
        var world = "World";
        return <span>{hello}, {world}</span>;
    }

});
```

```
$ eslint index.js

  ...
  6:28  error  A space is required after ','   comma-spacing

```

**Comment by shipstar** [comment]
This seems to have been [fixed upstream](https://github.com/eslint/eslint/issues/1916), but I'm still seeing it with the `babel-eslint` parser. Is this being treated as a duplicate of #31 or should it be reopened? (Or am I doing it wrong?)

**Comment by kevinrenskers** [comment]
I am having the same issue: I get this error with my JSX templates when using babel-eslint as my parser. No error when I don't use babel-eslint.

**Comment by hzoo** [comment]
Hmm this should of been fixed since I thought it was the same as #31. If not - what version of eslint/babel-eslint, and if it's not the latest can you test again?

If everything is updated, then can you give a code snippet so I can look into it?

**Comment by kevinrenskers** [comment]
Everything is updated. #31 is specifically about template strings and `${}`, maybe only that case was fixed?

```
export default React.createClass({
  displayName: 'HelloWorld',

  propTypes: {
    greeting: React.PropTypes.string,
    name: React.PropTypes.string
  },

  render() {
    return (
      <div>
        <h2>{this.props.greeting}, {this.props.name}!</h2>
      </div>
    );
  }
});
```

**Comment by hzoo** [comment]
Ok will look into it. I was thinking it was the same because of https://github.com/babel/babel-eslint/issues/31#issuecomment-77554119.

Also could you possibly give a smaller test case?

Is `var a = <h2>{this.props.greeting}, {this.props.name}!</h2>` not working?

**Comment by kevinrenskers** [comment]
Same problem: `A space is required after ',' comma-spacing`

**Comment by hzoo** [comment]
Ok thanks - can reproduce with `<h2>{this.props.greeting}, {this.props.name}!</h2>;` on an actual file. Just weird I'm not getting an error when writing the test for it

```js
it("template strings 2 #31", function () {
    verifyAndAssertMessages(
      "<h2>{this.props.greeting}, {this.props.name}!</h2>;",
      { "comma-spacing": 1 },
      []
    );
  });
```

**Comment by hzoo** [comment]
Haha I see why - I pushed https://github.com/babel/babel-eslint/commit/11f6298ba435c53d232d4f2d7ae78505d71f2468 which fixes this and it wasn't part of the latest release. So should be fixed in 3.1.5 which I'l do now.
